mne/viz/_figure.py

```diff
@@ -2295,6 +2074,88 @@ def _toggle_vline(self, visible):
         self.mne.vline_visible = visible
         self.canvas.draw_idle()
 
+    def close_all(self):
+        plt.close('all')
+
+    # workaround: plt.close() doesn't spawn close_event on Agg backend
+    # (check MPL github issue #18609; scheduled to be fixed by MPL 3.4)
+    def close_event(self, target=None):
+        """Force calling of the MPL figure close event."""
+        target = target or self
+        try:
+            target.canvas.close_event()
+        except ValueError:  # old mpl with Qt
+            pass  # pragma: no cover
+
+    def get_n_windows(self):
+        return len(plt.get_fignums())
+
+    def press_key(self, key, target=None):
+        target = target or self
+        target.canvas.key_press_event(key)
+
+    def fake_click(self, point, target=None, ax=None,
+                   xform='ax', button=1, kind='press'):
+        """Fake a click at a relative point within axes."""
+        target = target or self
+        ax = ax or self.mne.ax_main
+        if xform == 'ax':
+            x, y = ax.transAxes.transform_point(point)
+        elif xform == 'data':
+            x, y = ax.transData.transform_point(point)
+        else:
+            assert xform == 'pix'
+            x, y = point
+        if kind == 'press':
+            func = partial(target.canvas.button_press_event, x=x, y=y,
+                           button=button)
+        elif kind == 'release':
+            func = partial(target.canvas.button_release_event, x=x, y=y,
+                           button=button)
+        elif kind == 'motion':
+            func = partial(target.canvas.motion_notify_event, x=x, y=y)
+        else:
+            raise RuntimeError(f'{kind} is no suitabl kind for _fake_click')
+
+        func(guiEvent=None)
+
+    def click_ch_name(self, ch_index, button):
+        self.canvas.draw()
+        text = self.mne.ax_main.get_yticklabels()[ch_index]
+        bbox = text.get_window_extent()
+        x = bbox.intervalx.mean()
+        y = bbox.intervaly.mean()
+        self._fake_click((x, y), xform='pix', button=button)
+
+    def test_childs(self, key, attr):
```

Comment:
there should not be a testing function here.  Tests belong in `test_*.py` files inside `*/tests/` folders

Reply:
Yes, the naming is misleading. This is too something I just refactored in the first step from the tests. Maybe I can leave it as WIP? Because I think especially this test can be challenging to generalize between matplotlib and pyqtgraph.

Reply:
if there are a few tests that will only work for one backend or another, that's fine, but they should still live in a `test_*.py` file inside a `*/tests/` directory (and just not be parametrized to run under both backends).  Depending on how many such tests there are, they can either live in the existing `test_raw.py` or there could be separate `test_mpl_backend.py` and `test_pyqtgraph_backend.py` files to contain those tests.